crates/stackable-operator/src/client.rs

```diff
@@ -651,7 +657,25 @@ pub async fn initialize_operator(
         .context(InferKubeConfigSnafu)?;
     let default_namespace = kubeconfig.default_namespace.clone();
     let client = kube::Client::try_from(kubeconfig).context(CreateKubeClientSnafu)?;
-    let cluster_info = KubernetesClusterInfo::new(cluster_info_opts);
+
+    let local_cluster_info_opts = match cluster_info_opts.kubernetes_cluster_domain {
```

Comment:
I would prefer to not further enlarge this already gigantic file.
Please have a look at `KubernetesClusterInfo::new`, I'd say there is the correct place to add this, even with a todo marker :)
You need to make `KubernetesClusterInfo::new` fallible for that

Reply:
[bitteschön](https://github.com/stackabletech/operator-rs/pull/1068/commits/2e71f6751c344e4d0d46b05737c8dd75990247cf)